src/main/java/org/cubeville/cvgames/models/BaseGame.java

```diff
@@ -55,7 +61,61 @@ private void kickPlayerFromGame(Player p, boolean teleportToExit) {
 	public String getId() {
 		return id;
 	}
+	public List<Player> getSpectators() {
+		return spectators;
+	}
+	public void addSpectator(Player player) {
+		spectators.add(player);
+		player.teleport((Location) getVariable("spectator-spawn"));
+		player.setAllowFlight(true);
+		for (Player gp : getArena().getQueue().getPlayerSet()) {
+			gp.hidePlayer(CVGames.getInstance(), player);
+		}
+		hideSpectators(player);
+	}
+
+	public void removeSpectator(Player player) {
+		spectators.remove(player);
+		player.setAllowFlight(false);
+		for (Player gp : getArena().getQueue().getPlayerSet()) {
+			gp.showPlayer(CVGames.getInstance(), player);
+		}
+	}
+	public void showSpectators(Player player) {
+		for (Player sp : spectators) {
+			player.showPlayer(CVGames.getInstance(), sp);
+		}
+	}
+	public void hideSpectators(Player player) {
+		for (Player sp : spectators) {
+			player.hidePlayer(CVGames.getInstance(), sp);
+		}
+	}
 
+	public Inventory getPlayerCompassInventory(Player player, int page) {
+		List<ItemStack> contents = getPlayerCompassContents();
+		int invSize = (1 + (contents.size() / 9)) * 9;
+		if (invSize > 45) {
+			invSize = 54;
+			contents = contents.subList(0, 45);
+		}
+		Inventory inv = Bukkit.createInventory(player, invSize, "Player Compass");
+		int i = 0;
+		for (ItemStack item : contents) {
+			inv.setItem(i, item);
+			i++;
+		}
+		//TODO Page logic (If I really need to bother)
+		if (invSize > 45) {
+			for (i = invSize - 9; i < invSize; i++) {
+				if (i == 45) {
+
+				}
+			}
+		}
```

Comment:
Can you comment out this entire section so that it doesn't run for no reason at the end of this method?
(also you probably don't need to bother but a TODO wouldn't hurt for post release)